catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -945,10 +945,11 @@ public ResourceResponse retrieveResource(URI resourceUri,
                             requestedBytesToSkip,
                             responseBytesSkipped);
 
-                    return new ResourceResponseImpl(new ResourceImpl(new BufferedInputStream(
-                            resource.getInputStream()),
-                            resource.getMimeTypeValue(),
-                            FilenameUtils.getName(resource.getName())));
+                    try (BufferedInputStream stream = new BufferedInputStream(resource.getInputStream())) {
```

Comment:
If you close this stream, the client won't be able to download the resource passed through the source.